### PR TITLE
BREAKING: Split AssetsDetectionController

### PR DESCRIPTION
**Pull request by adonesky1**
**Split AssetsDetectionController into TokenDetectionController and CollectibleDetectionController**

This splits the AssetsDetectionController into a TokenDetectionController and CollectibleDetectionController in the same way we split [AssetController into TokensController and CollectiblesController](https://github.com/MetaMask/controllers/pull/518) without any functional changes to the API exposed by the original controller.

The immediate purpose of this PR is to enable integration of collectibles detection into the extension (part 1 found [here](https://github.com/MetaMask/metamask-extension/pull/12443)) without introducing duplicate token detection functionality (the extension has [its own TokenDetectionController](https://github.com/MetaMask/metamask-extension/blob/8e5acee42199aeabd46cf2cfb7a9dfe680fe540f/app/scripts/controllers/detect-tokens.js#L16) atm)
However a fast follow should be to make the TokenDetectionController shared code (merge the two controllers adding any missing functionality from either into the merged version).

- BREAKING:
  - Consumers of the AssetsDetectionController will have to now import both TokenDetectionController and CollectibleDetectionController and split up the calling of any methods accordingly (e.g. AssetDetectionController.detectAssets() => TokenDetectionController.detectTokens() and CollectibleDetectionController.detectCollectibles()).
     - Tested on Mobile with this branch: https://github.com/MetaMask/metamask-mobile/tree/split-asset-detection-controller (would need to add the changes to the controllers in this PR to test - `yarn build && yarn pack` in controllers and then `yarn add` the tarball to mobile repo.

### CODE REVIEW

**Review by ryanml — APPROVED** [review]
looks good from my end

File: src/assets/TokenDetectionController.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+      if (!tokensAddresses.includes(address)) {
+        tokensToDetect.push(address);
+      }
+    }
+    const sliceOfTokensToDetect = [];
+    sliceOfTokensToDetect[0] = tokensToDetect.slice(0, 1000);
+    sliceOfTokensToDetect[1] = tokensToDetect.slice(
+      1000,
+      tokensToDetect.length - 1,
+    );
```
Comment (NiranjanaBinoy) [inline src/assets/TokenDetectionController.ts:192]: Can we update this to the following code? This is the pattern we use in the extension detect-token.
    const sliceOfTokensToDetect = [
      tokensToDetect.slice(0, 1000),
      tokensToDetect.slice(1000, tokensToDetect.length - 1),
    ];

**Review by gantunesr — COMMENTED** [review]
LGTM

File: src/assets/TokenDetectionController.ts (hunk 1 above)
Comment (adonesky1) [reply src/assets/TokenDetectionController.ts:192]: This is a tiny semantic change and we can make the change in a followup PR to clean up code. We also will want to do an audit of difference between the extension token-detection controller and this one so we can handle that when we switch the extension to use of the controllers repo version.

**Review by hellbill — APPROVED** [review]

**Pull request merged**